libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -379,7 +381,9 @@ internal class MapboxTripSession(
         }
 
         private fun calculateRemainingWaypoints(tripStatus: TripStatus): Int {
-            val routeCoordinates = tripStatus.route?.routeOptions?.coordinatesList()
+            val routeCoordinates = tripStatus.route
+                ?.internalWaypoints()
+                ?.filter { it.isLegWaypoint() }
```

Comment:
This now doesn't include silent waypoints but it did in the past, doesn't this break the `waypointsCount - nextWaypointIndex` calculation below since NN does include silent waypoints in `tripStatus.navigationStatus.nextWaypointIndex`?

Reply:
it's actually breaking the logic, I'll fix it and add tests 👍